### ISSUE TITLE
Improper use of async/await could lead to deadlock

### ISSUE THREAD

**Issue by drub0y** [Microsoft/botbuilder-dotnet#1273]
This seems to have slipped under the radar when the work was being done to support OAuth, but the implementation of `BotFrameworkAdapter::CreateOAuthApiClientAsync` does some pretty naughty things with regards to async/await (or lack there of). 

The implementation currently looks as follows:

https://github.com/Microsoft/botbuilder-dotnet/blob/2629d57821149bdd3a596ebb4054f047fc97cff0/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L761-L784

The line that scares me is 779:

https://github.com/Microsoft/botbuilder-dotnet/blob/2629d57821149bdd3a596ebb4054f047fc97cff0/libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs#L779

The problem is, not only is it push work off to a separate task pool thread with the call to `Task::Run`, but then it's also not asynchronously awaiting that work and instead blocking the current thread, which was otherwise async, by calling `.Wait()` on the `Task` from the `Task::Run` call.

So...

  1. Why was `Task::Run` ever brought into play here?
  2. Why was `.Wait()` used instead of `await`?

Far as I can tell line 779 should just simply be refactored to:

```CSharp
  await oauthClient.SendEmulateOAuthCardsAsync(OAuthClient.EmulateOAuthCards).ConfigureAwait(false);
```

Happy to PR this fix in, but maybe the person who wrote the code wants to try and explain why they were doing this in the first place because maybe they think it's doing something it's not really doing.

**Comment by cleemullins** [comment]
Thanks @drub0y . I've tagged @Jeffders who (I believe) wrote this code. All things OAuth have been on his plate, so hopefully he'll chime in shortly.

**Comment by cleemullins** [comment]
Fixed by #1329
